test_core.TestMaskedArrayMethods.test_sort_flexible fails on ppc64 & s390x

### Reproducing code example:
Build NumPy and run the test suite in the Fedora build root. See full logs for [ppc64](https://koji.fedoraproject.org/koji/taskinfo?taskID=29010141) and [s390x](https://koji.fedoraproject.org/koji/taskinfo?taskID=29010143).

### Error message:

The result is the same for both arches:
```
E       AssertionError: 
E       Arrays are not equal
E       
E       (mismatch 85.7142857143%)
E        x: array([([1, 1],), ([1, 2],), ([2, 1],), ([2, 2],), ([3, 3],), ([3, 1],),
E              ([1, 0],)], dtype=[('v', '>i8', (2,))])
E        y: array([([1, 0],), ([1, 1],), ([1, 2],), ([2, 1],), ([2, 2],), ([3, 1],),
E              ([1, 3],)], dtype=[('v', '>i8', (2,))])
```

### Numpy/Python version information:
NumPy 1.15.0, Python 2.7

I believe both of these arches are big-endian, so this might be something relevant.

--- a/numpy/ma/tests/test_core.py
+++ b/numpy/ma/tests/test_core.py
@@ -3174,18 +3174,13 @@ class TestMaskedArrayMethods(object):
         assert_equal(test.mask, mask_first.mask)
 
         # Test sort on dtype with subarray (gh-8069)
+        # Just check that the sort does not error, structured array subarrays
+        # are treated as byte strings and that leads to differing behavior
+        # depending on endianess and `endwith`.
         dt = np.dtype([('v', int, 2)])
         a = a.view(dt)
-        mask_last = mask_last.view(dt)
-        mask_first = mask_first.view(dt)
-
         test = sort(a)
-        assert_equal(test, mask_last)
-        assert_equal(test.mask, mask_last.mask)
-
         test = sort(a, endwith=False)
-        assert_equal(test, mask_first)
-        assert_equal(test.mask, mask_first.mask)
 
     def test_argsort(self):
         # Test argsort

